File: csharp/src/Vista.SDK/GmodDto.cs
using System.Text.Json.Serialization;

namespace Vista.SDK;

internal sealed record GmodNodeDto(
    [property: JsonPropertyName("category")] string Category,
    [property: JsonPropertyName("type")] string Type,
    [property: JsonPropertyName("code")] string Code,
    [property: JsonPropertyName("name")] string Name,
    [property: JsonPropertyName("commonName")] string? CommonName,
    [property: JsonPropertyName("definition")] string? Definition,
    [property: JsonPropertyName("commonDefinition")] string? CommonDefinition,
    [property: JsonPropertyName("installSubstructure")] bool? InstallSubstructure,
    [property: JsonPropertyName("normalAssignmentNames")]
        IReadOnlyDictionary<string, string>? NormalAssignmentNames
);

internal sealed record GmodDto(
    [property: JsonPropertyName("visRelease")] string VisVersion,
    [property: JsonPropertyName("items")] GmodNodeDto[] Items,
    [property: JsonPropertyName("relations")] string[][] Relations
);

File: js/lib/types/GmodDto.ts
export interface GmodDto {
    visRelease: string;
    items: GmodNodeDto[];
    relations: [string, string][];
}

export interface GmodNodeDto {
    id?: string;
    category: string;
    type: string;
    code: string;
    name: string;
    commonName?: string | null;
    definition?: string | null;
    commonDefinition?: string | null;
    installSubstructure?: boolean | null;
    normalAssignmentNames?: { [key: string]: string };
}
